crates/iota/src/iota_commands.rs

```diff
@@ -751,8 +763,13 @@ async fn start(
     // note that this overrides the default configuration that is set when running
     // the genesis command, which sets data_ingestion_dir to None.
     #[cfg(feature = "indexer")]
-    if with_indexer.is_some() {
-        swarm_builder = swarm_builder.with_data_ingestion_dir(data_ingestion_path.clone());
+    if with_indexer.is_some() && data_ingestion_dir.is_none() {
+        data_ingestion_dir = Some(tempdir()?.into_path())
+    }
+
+    #[cfg(feature = "indexer")]
+    if let Some(ref dir) = data_ingestion_dir {
+        swarm_builder = swarm_builder.with_data_ingestion_dir(dir.clone());
```

Comment:
Should we throw an error if the user tries to use both?

Reply:
I don't think so, why shouldn't it be fine to have both at the same time?

Reply:
It says they are incompatable

Reply:
Where does it say this? I only see it being incompatible with `--no-full-node`